templates/core/core-dpl.yaml

```diff
@@ -85,6 +85,13 @@ spec:
               secretKeyRef:
                 name: "{{ template "harbor.jobservice" . }}"
                 key: JOBSERVICE_SECRET
+{{- if .Values.core.configureUserSettings }}
+          - name: CONFIG_OVERWRITE_JSON
```

Comment:
Thanks for doing this! I literally just now ran into the same issue that this PR solves!

Since we set `envFrom: - secretRef:`, I think that means CONFIG_OVERWRITE_JSON is always set as an environment variable for the pod. So wouldn't this block be unneeded?

Would that also mean we need to add `{{- if .Values.core.configureUserSettings }}` around the environment variable in the Secret? That way the environment variable isn't automatically set (by `envFrom`) even when the value is empty.

Reply:
Glad this will help!

Correct, we only want to define `CONFIG_OVERWRITE_JSON` if `.Values.core.configureUserSettings` is set. We need to add the if statement in the core secret. 

>Since we set `envFrom: - secretRef:`, I think that means `CONFIG_OVERWRITE_JSON` is always set as an environment variable for the container. So wouldn't this block be unneeded?

I don't understand what you mean here. Why is  `CONFIG_OVERWRITE_JSON` always set making this block unneeded?

Reply:
"envFrom" works by taking every key in the Secret and automatically inserting it as an environment variable in the container: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables

So no need to explicitly include the env var here; it is already implicitly included by "envFrom".

Reply:
> Thanks for doing this! I literally just now ran into the same issue that this PR solves!
> 
> Since we set `envFrom: - secretRef:`, I think that means CONFIG_OVERWRITE_JSON is always set as an environment variable for the container. So wouldn't this block be unneeded?
> 
> Would that also mean we need to add `{{- if .Values.core.configureUserSettings }}` around the environment variable in the Secret? That way the environment variable isn't automatically set (by `envFrom`) even when the value is empty.

The JSON may contain credentials, see above remarks. Preferably let's keep it "secret".

Reply:
> "envFrom" works by taking every key in the Secret and automatically inserting it as an environment variable in the container: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables
> 
> So no need to explicitly include the env var here; it is already implicitly included by "envFrom".

@mac-chaffee I understand now, you're talking about line [72-76](https://github.com/goharbor/harbor-helm/blob/562f4ae1e2934fceaf23a5957963e71dcb02cf06/templates/core/core-dpl.yaml#L72-L76). I'll remove the explicit inclusion of the environment variable and add the messing if statement to the core secret.

> The JSON may contain credentials, see above remarks. Preferably let's keep it "secret".

@janesser I think your concerns about the secret are still taken care of. The config will stay in the core secret. We just don't need to be explicitly including it in the core deployment. This is handled by the `envFrom` line linked above